include/pybind11/pytypes.h

```diff
@@ -319,6 +319,12 @@ PYBIND11_NAMESPACE_BEGIN(detail)
 inline std::string error_string();
 PYBIND11_NAMESPACE_END(detail)
 
+#ifdef __MINGW32__
```

Comment:
Having to repeat this worries me even more: when we refactor, will we have to create even more copies? How do we ensure the copies stay in sync?